docs/source/index.rst

```diff
@@ -0,0 +1,275 @@
+.. Flask-PluginEngine documentation master file, created by
+   sphinx-quickstart on Tue Mar 28 14:14:23 2017.
+   You can adapt this file completely to your liking, but it should at least
+   contain the root `toctree` directive.
+
+Flask-PluginEngine
+==================
+
+Flask-PluginEngine allows you to define and use different plugins in your Flask application
+
+
+==========
+Quickstart
+==========
+
+
+
+Installation
+------------
+
+You can install Flask-PluginEngine with the following command::
+
+    $ pip install Flask-PluginEngine
+
+
+Creating a plugin
+-----------------
+
+Creating a plugin is simple. First of all you need to create a class for your plugin inheriting from Plugin::
+
+    from flask_pluginengine import Plugin
+
+    class ExamplePlugin(Plugin):
+    """ExamplePlugin
+
+    Just an example plugin
+    """
+
+Second, you need to create a setup file. It's important to define a proper entry point to plugin's class. For example::
+
+    from setuptools import setup
+
+    setup(
+        name='Example-Plugin',
+        version='0.0.1',
+        py_modules=['example_plugin'],
+        entry_points={'example': {'example_plugin = example_plugin:ExamplePlugin'}}
+    )
+
+Now you can install the plugin running in console::
+
+    $ pip install -e .
+
+Current plugin
+--------------
+
+Very useful feature of Flask-PluginEngine is a ``current_plugin`` global value. It's a proxy to currently active plugin.
+
+Example plugin functionality
+----------------------------
+
+Now let's give our plugin some functionality to show how you can use the plugin. In this example we will render plugin's template from our application.
+
+Let's create the templates folder with a simple Jinja template. For example displaying plugin's title and description::
+
+    {% extends 'base.html' %}
+
+    {% block content %}
+        <h1>{{ plugin.title }}</h1>
+        <div>{{ plugin.description }}</div>
+    {% endblock %}
+
+If you documented plugin's class now you can access it as a ``name`` and ``description`` class properties.
+Notice that the template extends a base.html. We are going to use application's base template.
+
+Let's create a blueprint in our plugin::
+
+    plugin_blueprint = PluginBlueprint('example', __name__)
+
+In order to create a blueprint of a plugin we need to create PluginBlueprint instance.
+
+Then we can create a function rendering the template::
+
+    @plugin_blueprint.route('/plugin')
+    def hello_plugin():
+    return render_template('example_plugin:index.html', plugin=current_plugin)
+
+We can render any plugin template inside our application specifying name of the plugin before template name in ``render_template`` function, like above.
+We are also passing a current_plugin object to the template.
+
+
+Now in the plugin class we can create a method returning plugin's blueprint::
```

Comment:
"Now **, (note the comma)** in the plugin class we can create a method returning **the** plugin’s blueprint"